eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxow.hs

```diff
@@ -335,7 +347,7 @@ alonzoStyleWitness = do
   let utxo = utxosUtxo u
       txBody = tx ^. bodyTxL
       witsKeyHashes = keyHashWitnessesTxWits (tx ^. witsTxL)
-      scriptsProvided = getScriptsProvided utxo tx
+      scriptsProvided@(ScriptsProvided providedScriptsMap) = getScriptsProvided utxo tx
```

Comment:
`ScritpsProvided` do provide some type safety and most importantly it also provides information about the intent, so it is better to just pass through to `mkScriptIntegrity` function
```suggestion
      scriptsProvided = getScriptsProvided utxo tx
```